<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>LightingWidget</class>
 <widget class="QWidget" name="LightingWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>880</width>
    <height>1410</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Shader Widget</string>
  </property>
  <layout class="QVBoxLayout">
   <item>
    <widget class="QCheckBox" name="lightposition">
     <property name="text">
      <string>Light position</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="DirectionVectorWidget" name="lightpositionW">
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QCheckBox" name="applyemissiveD">
     <property name="text">
      <string>Emissive</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QCheckBox" name="peel">
     <property name="text">
      <string>Reveal</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QFrame" name="peelW">
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="1" column="0">
       <widget class="QLabel" name="label_12">
        <property name="text">
         <string>frontface</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label_17">
        <property name="text">
         <string>type</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_13">
        <property name="text">
         <string>backface</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_16">
        <property name="text">
         <string>mix</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QSlider" name="peelmix_">
        <property name="minimumSize">
         <size>
          <width>200</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximum">
         <number>100</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="tickPosition">
         <enum>QSlider::TicksAbove</enum>
        </property>
        <property name="tickInterval">
         <number>10</number>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSlider" name="peelmin_">
        <property name="minimumSize">
         <size>
          <width>200</width>
          <height>0</height>
         </size>
        </property>
        <property name="minimum">
         <number>-110</number>
        </property>
        <property name="maximum">
         <number>110</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="tickPosition">
         <enum>QSlider::TicksAbove</enum>
        </property>
        <property name="tickInterval">
         <number>10</number>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QSlider" name="peelmax_">
        <property name="minimumSize">
         <size>
          <width>200</width>
          <height>0</height>
         </size>
        </property>
        <property name="minimum">
         <number>-110</number>
        </property>
        <property name="maximum">
         <number>110</number>
        </property>
        <property name="value">
         <number>0</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="tickPosition">
         <enum>QSlider::TicksAbove</enum>
        </property>
        <property name="tickInterval">
         <number>10</number>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QSpinBox" name="peelmin_SB"/>
      </item>
      <item row="2" column="2">
       <widget class="QSpinBox" name="peelmax_SB"/>
      </item>
      <item row="3" column="2">
       <widget class="QSpinBox" name="peelmix_SB"/>
      </item>
      <item row="0" column="1" colspan="2">
       <widget class="QComboBox" name="peeltype">
        <property name="editable">
         <bool>false</bool>
        </property>
        <item>
         <property name="text">
          <string>Peel</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Keep Inside</string>
         </property>
        </item>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QCheckBox" name="applylighting">
     <property name="text">
      <string>Lighting</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QFrame" name="applylightingW">
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="1">
       <widget class="QSlider" name="ambient_">
        <property name="minimumSize">
         <size>
          <width>200</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximum">
         <number>10</number>
        </property>
        <property name="pageStep">
         <number>1</number>
        </property>
        <property name="value">
         <number>10</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="tickPosition">
         <enum>QSlider::TicksAbove</enum>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>hilite size</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QSlider" name="specular_">
        <property name="minimumSize">
         <size>
          <width>200</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximum">
         <number>10</number>
        </property>
        <property name="pageStep">
         <number>1</number>
        </property>
        <property name="value">
         <number>10</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="tickPosition">
         <enum>QSlider::TicksAbove</enum>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QSlider" name="specularcoeff_">
        <property name="minimumSize">
         <size>
          <width>200</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximum">
         <number>9</number>
        </property>
        <property name="pageStep">
         <number>1</number>
        </property>
        <property name="value">
         <number>2</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="tickPosition">
         <enum>QSlider::TicksAbove</enum>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>diffuse</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSlider" name="diffuse_">
        <property name="minimumSize">
         <size>
          <width>200</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximum">
         <number>10</number>
        </property>
        <property name="pageStep">
         <number>1</number>
        </property>
        <property name="value">
         <number>0</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="tickPosition">
         <enum>QSlider::TicksAbove</enum>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>ambient</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>specular</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QSpinBox" name="ambient_SB"/>
      </item>
      <item row="1" column="2">
       <widget class="QSpinBox" name="diffuse_SB"/>
      </item>
      <item row="2" column="2">
       <widget class="QSpinBox" name="specular_SB"/>
      </item>
      <item row="3" column="2">
       <widget class="QSpinBox" name="specularcoeff_SB"/>
      </item>
      <item row="5" column="0" colspan="3">
       <widget class="QFrame" name="applyshadowW">
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <layout class="QVBoxLayout" name="_2">
         <item>
          <widget class="QWidget" name="widget" native="true">
           <layout class="QGridLayout" name="gridLayout_3">
            <item row="1" column="1">
             <widget class="QSlider" name="shadowscale_">
              <property name="minimumSize">
               <size>
                <width>200</width>
                <height>0</height>
               </size>
              </property>
              <property name="maximum">
               <number>8</number>
              </property>
              <property name="pageStep">
               <number>1</number>
              </property>
              <property name="value">
               <number>5</number>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="tickPosition">
               <enum>QSlider::TicksAbove</enum>
              </property>
             </widget>
            </item>
            <item row="0" column="0">
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>blur</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSlider" name="shadowblur_">
              <property name="minimumSize">
               <size>
                <width>200</width>
                <height>0</height>
               </size>
              </property>
              <property name="maximum">
               <number>5</number>
              </property>
              <property name="pageStep">
               <number>1</number>
              </property>
              <property name="value">
               <number>1</number>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="tickPosition">
               <enum>QSlider::TicksAbove</enum>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_6">
              <property name="text">
               <string>buffer size</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QSlider" name="shadowcontrast_">
              <property name="minimumSize">
               <size>
                <width>200</width>
                <height>0</height>
               </size>
              </property>
              <property name="maximum">
               <number>10</number>
              </property>
              <property name="pageStep">
               <number>1</number>
              </property>
              <property name="value">
               <number>5</number>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="tickPosition">
               <enum>QSlider::TicksAbove</enum>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_7">
              <property name="text">
               <string>contrast</string>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QLabel" name="label_8">
              <property name="text">
               <string>cone angle</string>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="QSlider" name="shadowfov_">
              <property name="minimumSize">
               <size>
                <width>200</width>
                <height>0</height>
               </size>
              </property>
              <property name="maximum">
               <number>10</number>
              </property>
              <property name="pageStep">
               <number>2</number>
              </property>
              <property name="value">
               <number>5</number>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="tickPosition">
               <enum>QSlider::TicksAbove</enum>
              </property>
             </widget>
            </item>
            <item row="0" column="2">
             <widget class="QSpinBox" name="shadowblur_SB"/>
            </item>
            <item row="1" column="2">
             <widget class="QSpinBox" name="shadowscale_SB"/>
            </item>
            <item row="2" column="2">
             <widget class="QSpinBox" name="shadowcontrast_SB"/>
            </item>
            <item row="3" column="2">
             <widget class="QSpinBox" name="shadowfov_SB"/>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="applycoloredshadow">
           <property name="text">
            <string>Shadow Color Attenuation</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QFrame" name="applycoloredshadowW">
           <property name="frameShape">
            <enum>QFrame::StyledPanel</enum>
           </property>
           <layout class="QGridLayout" name="gridLayout_4">
            <item row="3" column="1">
             <widget class="QSlider" name="blue_">
              <property name="minimumSize">
               <size>
                <width>200</width>
                <height>0</height>
               </size>
              </property>
              <property name="maximum">
               <number>100</number>
              </property>
              <property name="pageStep">
               <number>5</number>
              </property>
              <property name="value">
               <number>50</number>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="tickPosition">
               <enum>QSlider::TicksAbove</enum>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QLabel" name="label_11">
              <property name="text">
               <string>blue</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QSlider" name="green_">
              <property name="minimumSize">
               <size>
                <width>200</width>
                <height>0</height>
               </size>
              </property>
              <property name="maximum">
               <number>100</number>
              </property>
              <property name="pageStep">
               <number>5</number>
              </property>
              <property name="value">
               <number>50</number>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="tickPosition">
               <enum>QSlider::TicksAbove</enum>
              </property>
             </widget>
            </item>
            <item row="0" column="0" colspan="2">
             <widget class="QCheckBox" name="linkcolors">
              <property name="text">
               <string>Link color sliders</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QSlider" name="red_">
              <property name="minimumSize">
               <size>
                <width>200</width>
                <height>0</height>
               </size>
              </property>
              <property name="maximum">
               <number>100</number>
              </property>
              <property name="pageStep">
               <number>5</number>
              </property>
              <property name="value">
               <number>50</number>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="tickPosition">
               <enum>QSlider::TicksAbove</enum>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_9">
              <property name="text">
               <string>red</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_10">
              <property name="text">
               <string>green</string>
              </property>
             </widget>
            </item>
            <item row="1" column="2">
             <widget class="QSpinBox" name="red_SB"/>
            </item>
            <item row="2" column="2">
             <widget class="QSpinBox" name="green_SB"/>
            </item>
            <item row="3" column="2">
             <widget class="QSpinBox" name="blue_SB"/>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="applybackplane">
           <property name="text">
            <string>Backplane</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QFrame" name="applybackplaneW">
           <property name="frameShape">
            <enum>QFrame::StyledPanel</enum>
           </property>
           <layout class="QGridLayout" name="gridLayout_5">
            <item row="1" column="1">
             <widget class="QSlider" name="backplanecontrast_">
              <property name="minimumSize">
               <size>
                <width>200</width>
                <height>0</height>
               </size>
              </property>
              <property name="maximum">
               <number>10</number>
              </property>
              <property name="pageStep">
               <number>2</number>
              </property>
              <property name="value">
               <number>10</number>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="tickPosition">
               <enum>QSlider::TicksAbove</enum>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_15">
              <property name="text">
               <string>contrast</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSlider" name="backplaneshadowscale_">
              <property name="minimumSize">
               <size>
                <width>200</width>
                <height>0</height>
               </size>
              </property>
              <property name="minimum">
               <number>5</number>
              </property>
              <property name="maximum">
               <number>15</number>
              </property>
              <property name="pageStep">
               <number>2</number>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="tickPosition">
               <enum>QSlider::TicksAbove</enum>
              </property>
             </widget>
            </item>
            <item row="0" column="0">
             <widget class="QLabel" name="label_14">
              <property name="text">
               <string>scale</string>
              </property>
             </widget>
            </item>
            <item row="0" column="2">
             <widget class="QSpinBox" name="backplaneshadowscale_SB"/>
            </item>
            <item row="1" column="2">
             <widget class="QSpinBox" name="backplanecontrast_SB"/>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="4" column="0" colspan="3">
       <widget class="QCheckBox" name="applyshadow">
        <property name="text">
         <string>Shadows</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::MinimumExpanding</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>0</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>DirectionVectorWidget</class>
   <extends>QFrame</extends>
   <header>directionvectorwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>peel</sender>
   <signal>toggled(bool)</signal>
   <receiver>peelW</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>72</x>
     <y>105</y>
    </hint>
    <hint type="destinationlabel">
     <x>49</x>
     <y>206</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>applylighting</sender>
   <signal>toggled(bool)</signal>
   <receiver>applylightingW</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>100</x>
     <y>281</y>
    </hint>
    <hint type="destinationlabel">
     <x>84</x>
     <y>774</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>applyshadow</sender>
   <signal>toggled(bool)</signal>
   <receiver>applyshadowW</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>117</x>
     <y>448</y>
    </hint>
    <hint type="destinationlabel">
     <x>110</x>
     <y>901</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>applycoloredshadow</sender>
   <signal>toggled(bool)</signal>
   <receiver>applycoloredshadowW</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>152</x>
     <y>633</y>
    </hint>
    <hint type="destinationlabel">
     <x>144</x>
     <y>778</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>applybackplane</sender>
   <signal>toggled(bool)</signal>
   <receiver>applybackplaneW</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>148</x>
     <y>807</y>
    </hint>
    <hint type="destinationlabel">
     <x>148</x>
     <y>891</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lightposition</sender>
   <signal>toggled(bool)</signal>
   <receiver>lightpositionW</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>42</x>
     <y>23</y>
    </hint>
    <hint type="destinationlabel">
     <x>63</x>
     <y>47</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
